lib/rspec/its.rb

```diff
@@ -123,6 +141,14 @@ def is_expected
         end
         alias_method :are_expected, :is_expected
 
+        def will(matcher=nil, message=nil)
+          expect { __its_subject }.to matcher, message
```

Comment:
Here we need to check that matcher supports block expectations, not all of them do, it should be relatively simple e.g:
```
raise ArgumentError, "`will` only supports block expectations" unless matcher.supports_block_expectations?
```

And the same for `will_not`

Reply:
👍